core/types/calls.cc

```diff
@@ -996,6 +996,11 @@ DispatchResult dispatchCallSymbol(const GlobalState &gs, const DispatchArgs &arg
     // the send, assume that the last argument is an implicit keyword args hash.
     bool implicitKwsplat = false;
     if (ait != aPosEnd && hasKwargs && args.args.size() == args.numPosArgs) {
+        if (Types::isSubType(gs, (*ait)->type, Types::untypedUntracked())) {
+            if (auto e = gs.beginError(args.argLoc(args.args.size() - 1), errors::Infer::UntypedKeywordArg)) {
+                e.setHeader("Positional argument being interpreted as keyword argument is `T.untyped`");
+            }
+        }
```

Comment:
Should we only report this if `ruby3KeywordArgs` is not enabled?

In that case, the untyped error is somewhat redundant, and might serve to confuse or disguise the root error.